### ISSUE TITLE
[Missing SNMP Polling Profile] Sophos XGS 5500

### ISSUE THREAD

**Issue by edwardflanders** [kentik/snmp-profiles#250]
[NOTE]: # ( ^^ Please add the Device Vendor and Model in the title above. ^^ )

Device Vendor: Sophos

Device Model: XGS5500

[NOTE]: # (If you do not know your SysOID, it can be gathered from the SNMP walk you provide at the end of this issue. It is the value of this OID in the SYSTEM-MIB: `1.3.6.1.2.1.1.2.0`)

System Object Identifier (SysOID):  .1.3.6.1.4.1.2604.5

List of critical metrics that you expect in the profile (does not need to be all-inclusive, just a general idea of your expectations):

 - CPU Utilization
 - Memory Utilization
 - Interface Metrics
 - Broadcast Packets
 - Multicast Packets
 

**Please provide a [public gist](https://gist.github.com/) with a sanitized** *(sensitive information removed)* **SNMP walk of the device. Providing MIB files is not a substitute for an SNMP walk and failure to provide one here will result in delays for closing this issue. The output of these commands shows the "true story" of what a device will respond to in SNMP and is used to ensure that our profiles are efficient and accurate for your devices.**

Gist Link: https://gist.github.com/edwardflanders/fd2373efd4e3a3e33938890e5cbf8b06

**Comment by thezackm** [comment]
@edwardflanders - thank you for your submission, can you please validate the results of this SNMP walk? It looks like the walk timed out around 15k results in the IP-MIB's [inetCidrRouteTable](http://oid-info.com/get/1.3.6.1.2.1.4.24.7) (`.1.3.6.1.2.1.4.24.7.*`). For context; the IF-MIB where most interface performance metrics are would be in the `.1.3.6.1.2.1.31.*` root and anything specific to the vendor would be even further down with a prefix of `.1.3.6.1.4.*`. So the walk you've submitted has the marks of a device that just has a LOT of SNMP results and needs more time to process through them. 

You can try to directly poll specific MIBs or the root OIDs with walk commands like this and see if you get better results too:

```
snmpwalk -v 2c -t 10 -On -c $COMMUNITY_STRING $IP_ADDRESS .1.3.6.1.2.1.31 >> if-mib_walk.out

snmpwalk -v 2c -t 10 -On -c $COMMUNITY_STRING $IP_ADDRESS .1.3.6.1.4 >> vendor-root_walk.out
```

Without these results, we cannot know what OIDs this device will work with.

**Comment by edwardflanders** [comment]
@Zack Mutchler

Thanks for your message.  I’ve run both of the walks suggested below and hopefully you can pull what’s needed from them. 

Please find the gist urls here:

https://gist.github.com/edwardflanders/2175ead64b441141772a4964bb7ad407
https://gist.github.com/edwardflanders/6ebe7c8471e02408c949cc5fcbf025c9

Many thanks

Ed

**Comment by thezackm** [comment]
Thanks @edwardflanders! That looks great! 

The one item I am not seeing is CPU utilization in the Sophos MIB on that 2nd walk. Sometimes that's kept in the `HOST-RESOURCES-MIB`; but it's best if you contact the vendor if possible to get their recommendation on the best OID to use for this to be sure. 

You can definitely try to walk that MIB and we can see if the results match up with what you see on your device though:

```
snmpwalk -v 2c -t 10 -On -c $COMMUNITY_STRING $IP_ADDRESS .1.3.6.1.2.1.25 >> host-resources-mib_walk.out
```

**Comment by edwardflanders** [comment]
I've double checked some documentation and this is the oid for cpu: 1.3.6.1.2.1.25.3.3.1.2.196608

I can see this is also in the host-resources walk as you would expect.

please see the gist link as well: https://gist.github.com/edwardflanders/07fd8c927f5f96fbe40d07bf12825432

Incidentally I do have some other xg devices which I will need to pull snmp data from but I understand the oids are version specific not hardware from what I have read and they all have the same sysobjectID

Thanks

**Comment by edwardflanders** [comment]
Hi there,


Any update on this one.  Commenting as I received a stale notice from GitHub and don't want this closed automatically.

Regards
Ed

**Comment by thezackm** [comment]
addressed in PR #305; note that we need to see this data in our UI before we can built the default experience around it. Once this PR is merged and a new container image is available please pull that down and run a fresh discovery against your devices. When you have successfully started collecting data please send me a DM on the [New Relic Explorer's Hub](https://discuss.newrelic.com/), username `zackm` with your account ID and we'll get that sorted ASAP.

*alternatively you can share your email address you use to login to New Relic here, but using the Explorer's Hub will maintain your privacy, it's your choice*

**Comment by edwardflanders** [comment]
@Zack ***@***.***>

Thanks for your message.  I’ve run both of the walks suggested below and hopefully you can pull what’s needed from them.

Please find the gist urls here:

https://gist.github.com/edwardflanders/2175ead64b441141772a4964bb7ad407
https://gist.github.com/edwardflanders/6ebe7c8471e02408c949cc5fcbf025c9

Many thanks

Ed


Edward Flanders
Network Engineer
Leeds Road
Huddersfield
HD2 1UA
West Yorkshire

Telephone:
Email: ***@***.***
Fax: 0844 774 6303

[Visit us at http://www.buyitdirect.co.uk]<http://www.buyitdirect.co.uk>

Our Brands:
[http://www.buyitdirect.co.uk]<http://www.buyitdirect.co.uk>

Opening Times and Contact details<http://www.buyitdirect.co.uk/content/contact.asp>

Buy It Direct Ltd is a limited company registered in England. Registered number: 04171412. Registered office: Unit A Trident Business Park, Leeds Road,Huddersfield, West Yorkshire, HD2 1UA.

**Comment by thezackm** [comment]
@edwardflanders - this issue was closed with PR #305 in May; have you tested the latest image against your firewalls and found metrics to be missing?
